src/sentry/event_manager.py

```diff
@@ -1139,12 +1142,7 @@ def _save_aggregate(event, hashes, release, metadata, received_timestamp, **kwar
 
     is_new = False
 
-    if root_hierarchical_hash is not None:
-        new_hashes = []
```

Comment:
`root_hierarchical_hash` is also set when the hierarchical `GroupHash` is new, which means that a possible secondary group hash won't make it into `new_hashes` here.